### PR TITLE
Implement AOS-46 Idempotent storager delete operation

### PR DESCRIPTION
**Pull request by JinnyYi**
Idempotent storager delete operation in ref: https://github.com/aos-dev/go-storage/issues/554.

### CODE REVIEW

File: multiparter.go [hunk 1]
```diff
@@ -65,6 +65,23 @@ func TestMultiparter(t *testing.T, store types.Storager) {
...
+			m, _ := store.(types.Multiparter)
+
+			path := uuid.New().String()
+			o, err := m.CreateMultipart(path)
+			if err != nil {
+				t.Error(err)
+			}
+
+			for i := 0; i < 2; i++ {
+				err = store.Delete(path, pairs.WithMultipartID(o.MustGetMultipartID()))
```
Comment (Xuanwo) [inline multiparter.go]: Please call `store.Delete` directly.

And check them twice:

```
Convey("The first returning error should be nil")
Convey("The second returning error also should be nil")
```

File: storager.go [hunk 2]
```diff
@@ -197,6 +197,35 @@ func TestStorager(t *testing.T, store types.Storager) {
 			})
 		})
 
+		Convey("When Delete a file twice", func() {
```
Comment (Xuanwo) [inline storager.go]: We can test this logic in the `Delete` test instead of a new test case.

File: multiparter.go (hunk 1 above)
Comment (JinnyYi) [reply multiparter.go]: Does call `store.Delete` directly mean call `store.Delete` without the secondary parameter? If so, maybe we cannot get the `MultipartIDNotExist` liked error returned by `AbortMultipartUpload`.

Comment (Xuanwo) [reply multiparter.go]: No, call directly means don't use `for i := 0; i < 2; i++` here.

File: appender.go [hunk 3]
```diff
@@ -45,7 +45,7 @@ func TestAppender(t *testing.T, store types.Storager) {
 			})
 		})
 
-		Convey("When Delete", func() {
+		Convey("When Delete twice", func() {
```
Comment (Xuanwo) [inline appender.go]: We don't need to mark `twice` here.

File: multiparter.go [hunk 4]
```diff
@@ -50,7 +50,7 @@ func TestMultiparter(t *testing.T, store types.Storager) {
 			})
 		})
 
-		Convey("When Delete with multipart id", func() {
+		Convey("When Delete with multipart id twice", func() {
```
Comment (Xuanwo) [inline multiparter.go]: ditto

File: storager.go [hunk 5]
```diff
@@ -170,7 +170,7 @@ func TestStorager(t *testing.T, store types.Storager) {
 			})
 		})
 
-		Convey("When Delete a file", func() {
+		Convey("When Delete a file twice", func() {
```
Comment (Xuanwo) [inline storager.go]: ditto

File: storager.go [hunk 6]
```diff
@@ -185,11 +185,24 @@ func TestStorager(t *testing.T, store types.Storager) {
 
 			err = store.Delete(path)
 
-			Convey("The error should be nil", func() {
+			Convey("The first returning error should be nil", func() {
+				So(err, ShouldBeNil)
+			})
+
+			Convey("The first Stat should get nil Object and ObjectNotFound error", func() {
```
Comment (Xuanwo) [inline storager.go]: `Stat` only needs to test once.

**Pull request merged**